auth/token_issuer_test.go

```diff
@@ -112,3 +115,76 @@ func TestCreateToken(t *testing.T) {
 		}
 	}
 }
+
+func TestTTL(t *testing.T) {
+	e := &ldap.Entry{
+		DN: "some-dn",
+	}
+
+	cases := []struct {
+		TTL time.Duration
+		expectedTTL time.Duration
+	} {
+		{
+			expectedTTL: 24 * time.Hour,
+		},
+		{
+			TTL: 5 * time.Second,
+			expectedTTL: 5* time.Second,
+		},
+	}
+
+	for i, c := range cases {
+		lti := LDAPTokenIssuer{
+			LDAPServer: "some-ldap-server",
+			TTL: c.TTL,
+		}
+
+		tok := lti.createToken(e)
+		now := time.Now().UnixNano() / int64(time.Millisecond)
+		expectedExpiration := now + int64(time.Duration(c.TTL) / time.Millisecond)
+
+		if tok.Expiration > expectedExpiration {
+			t.Errorf("Case: %d. Expiration expected: %d, got: %d", i, tok.Expiration, expectedExpiration)
+		}
+	}
+}
+
+func TestTokenExpired(t *testing.T) {
+	e := &ldap.Entry{
+		DN: "some-dn",
+	}
+
+	cases := []struct {
+		TTL time.Duration
+		sleep time.Duration
+		expectedTokenExpired bool
+	} {
+		{
+			TTL: 200 * time.Millisecond,
+			sleep: 100 * time.Millisecond,
+			expectedTokenExpired: false,
+		},
+		{
+			TTL: 200 * time.Millisecond,
+			sleep: 300 * time.Millisecond,
+			expectedTokenExpired: true,
+		},
+	}
+
+	for i, c := range cases {
+		lti := LDAPTokenIssuer{
+			LDAPServer: "some-ldap-server",
+			TTL: c.TTL,
+		}
+
+		tok := lti.createToken(e)
+
+		time.Sleep(c.sleep)
```

Comment:
Normally for this sort of test one would mock out the clock through an interface. That would also allow the test to assert that the token doesn't expire too early.

Reply:
golang by default don't provide a fake clock, and using a different interface will need more changes. 

in the interest of time, I will add an issue for it to be addressed later. does that sound ok? @johngmyers

Reply:
ok to address later. You'd just want to define an interface matching `time.Now()` and use time.Now() for the default constructed verifier.